src/test/e2e/psp.feature.ts

```diff
@@ -0,0 +1,263 @@
+import fs from 'fs';
+import path from 'path';
+import { mockReadFileSync } from '../mock-utils';
+import { ContractFactory, Contract } from 'ethers';
+import * as hre from 'hardhat';
+import * as abi from '@api3/airnode-abi';
+import * as node from '@api3/airnode-node';
+import * as psp from '../../handlers/psp';
+import * as config from '../../config';
+import { buildAirnodeConfig, buildAirkeeperConfig, buildLocalConfig } from '../config/config';
+import { PROTOCOL_ID_PSP } from '../../constants';
+
+// Jest version 27 has a bug where jest.setTimeout does not work correctly inside describe or test blocks
+// https://github.com/facebook/jest/issues/11607
+jest.setTimeout(30_000);
+
+describe('PSP', () => {
+  process.env = Object.assign(process.env, {
+    CLOUD_PROVIDER: 'local',
+    STAGE: 'dev',
+  });
+
+  const airnodeConfig = buildAirnodeConfig();
+  const airkeeperConfig = buildAirkeeperConfig();
+  const localConfig = buildLocalConfig();
+
+  const provider = new hre.ethers.providers.JsonRpcProvider('http://127.0.0.1:8545');
+
+  const roles = {
+    deployer: new hre.ethers.Wallet(localConfig.privateKeys.deployer).connect(provider),
+    manager: new hre.ethers.Wallet(localConfig.privateKeys.manager).connect(provider),
+    sponsor: new hre.ethers.Wallet(localConfig.privateKeys.sponsor).connect(provider),
+    randomPerson: new hre.ethers.Wallet(localConfig.privateKeys.randomPerson).connect(provider),
+  };
+
+  const dapiServerAdminRoleDescription = 'DapiServer admin';
+  let accessControlRegistryAbi;
+  let accessControlRegistryFactory: ContractFactory;
+  let accessControlRegistry: Contract;
+  let airnodeProtocolAbi;
+  let airnodeProtocolFactory: ContractFactory;
+  let airnodeProtocol: Contract;
+  let dapiServerAbi;
+  let dapiServerFactory: ContractFactory;
+  let dapiServer: Contract;
+
+  beforeEach(async () => {
+    //Reset the local hardhat network state for each test to keep the deployed Airnode and DapiServer contract addresses
+    //the same as the config files
+    await hre.network.provider.send('hardhat_reset');
```

Comment:
This took me way too long to figure this out, but there were a couple issues with the original setup:

1. The (dapiServer, airnodeProtocol) were re-deployed for each test and since the hardhat network was running, those had different addresses for each test. Then the **DapiServer** and **AirnodeRrp** addressed in the config files didn't match and the values were not updated (even though the handler finishes correctly).

I had to make a few modifications to get this hardhat_reset to work (namely copy the hardhat.config.ts to the root after install), otherwise I got an error that he project 'was not a hardhat project'.

2. `mockReadFileSync` does not seem to be working for this (even when I change it to mockImplementation), but it errors out saying that the airnode config cannot be found. So I changed the loadConfig functions to be mocked instead.

3. I tried to derive the **beaconid** with the airnodeProtocol address and the templateId, but it ends up giving me a different id than getting it from the dapiServer contract with `subscriptionIdToBeaconId()`. I'm not sure why that is.

Reply:
I am not sure about the 3), that seems like a thing which might bite us in production. Can you maybe check why is that? (not necessarily in this PR)

Reply:
@vponline were you able to figure 3 out?

Reply:
@Siegrift  @acenolaza  This was my bad, I was trying to derive it with the airnode protocol address instead of the airnode address from the config... :facepalm: I updated `beaconId` to be derived now.